template/src/ontology/run.sh.jinja2

```diff
@@ -23,10 +30,20 @@ if [ x$ODK_DEBUG = xyes ]; then
     TIMECMD="/usr/bin/time -f ### DEBUG STATS ###\nElapsed time: %E\nPeak memory: %M kb"
 fi
 
-docker run -v $PWD/../../:/work -w /work/src/ontology -e ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS" -e JAVA_OPTS="$ODK_JAVA_OPTS" {% if project.use_external_date is sameas True %}-e TODAY=`date +%Y-%m-%d` {% endif %}--rm -ti obolibrary/$IMAGE $TIMECMD "$@"
+VOLUME_BIND=$PWD/../../:/work
+WORK_DIR=/work/src/ontology
+
+if [ -n "$USE_SINGULARITY" ]; then
+    export ROBOT_JAVA_ARGS="$ODK_JAVA_OPTS"
+    export JAVA_OPTS="$ODK_JAVA_OPTS"
+    {% if project.use_external_date is sameas True %}export TODAY=$(date +%Y-%m-%d){% endif %}
+    singularity exec --bind $VOLUME_BIND -W $WORK_DIR docker://obolibrary/$IMAGE:$TAG $TIMECMD "$@"
```

Comment:
Have you tested this?

Reply:
yes, the singularity call is working like this on the EBI cluster.

Reply:
I haven't tested the ansible part though.